src/app/src/main/kotlin/org/imperial/mrc/hint/AppStart.kt

```diff
@@ -2,10 +2,30 @@ package org.imperial.mrc.hint
 
 import org.springframework.boot.SpringApplication
 import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.context.annotation.Configuration
+import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+import org.springframework.web.servlet.config.annotation.EnableWebMvc
+import org.springframework.web.servlet.resource.EncodedResourceResolver
+import org.springframework.web.servlet.resource.PathResourceResolver
+
 
 @SpringBootApplication
 open class HintApplication
 
 fun main(args: Array<String>) {
     SpringApplication.run(HintApplication::class.java, *args)
 }
+
+@Configuration
+@EnableWebMvc
+open class MvcConfig : WebMvcConfigurer {
+    override fun addResourceHandlers(registry: ResourceHandlerRegistry) {
+        registry.addResourceHandler("/public/**")
+                .addResourceLocations("file:/static/public/", "file:static/public/")
+                .resourceChain(true)
+                .addResolver(EncodedResourceResolver())
+                .addResolver(PathResourceResolver())
```

Comment:
This first resolver returns gzipped content when requested, the second is the default resolver that any request can fallback to.